### PR TITLE
chore: refactor golden tests for autopopulation feature

### PR DESCRIPTION
**Pull request by alicejli**
As a clean-up follow up to https://github.com/googleapis/sdk-platform-java/pull/2353, this PR refactors the unit golden tests for the autopopulation feature to a separate, new proto called `auto_populate_field_testing.proto`.

### CODE REVIEW

File: gapic-generator-java/src/test/proto/auto_populate_field_testing.proto [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    option (google.api.http) = {
+      post: "/v1beta1/AutoPopulateFieldTesting:collect"
+      body: "*"
+    };
+  }
+
+  // This bidirectional streaming method, upon receiving a request on the stream, will pass the same
+  // content back on the stream. This method should not have any autopopulated fields.
+  rpc AutoPopulateFieldTestingChat(stream AutoPopulateFieldTestingEchoRequest) returns (stream AutoPopulateFieldTestingEchoResponse);
+}
```
Comment (lqiu96) [inline gapic-generator-java/src/test/proto/auto_populate_field_testing.proto:68]: These don't have autopopulated fields b/c they're streaming RPCs, right?

**Comment by lqiu96** [comment]
Can you also check if there should be goldens for ServiceStubClassComposerTest in grpcrest?

**Comment by lqiu96** [comment]
Besides the two questions above, LGTM.

File: gapic-generator-java/src/test/proto/auto_populate_field_testing.proto (hunk 1 above)
Comment (alicejli) [reply gapic-generator-java/src/test/proto/auto_populate_field_testing.proto:68]: Yup, that's correct.

**Comment by alicejli** [comment]
> Can you also check if there should be goldens for ServiceStubClassComposerTest in grpcrest?

I've added in the goldens for them. Thanks!

File: gapic-generator-java/src/test/proto/auto_populate_field_testing.proto (hunk 1 above)
Comment (lqiu96) [reply gapic-generator-java/src/test/proto/auto_populate_field_testing.proto:68]: Could you add a quick blurb that this is b/c it is a streaming RPC? I saw that it has `AutoPopulateFieldTestingEchoRequest` and forgot about streaming.

**Comment by lqiu96** [comment]
> > Can you also check if there should be goldens for ServiceStubClassComposerTest in grpcrest?
> 
> I've added in the goldens for them. Thanks!

I took a look at the grpc+rest composers and we don't think that the auto populate PR impacts those composers. We probably don't need goldens there, so you can remove those golden cases.

**Review by lqiu96 — APPROVED** [review]
LGTM, if you can make the two quick changes. Thanks!

**Pull request merged**